text/0000-feature-descriptions-doc-cfg.md

```diff
@@ -0,0 +1,208 @@
+- Feature Name: feature-descriptions-doc-cfg
+- Start Date: 2023-04-14
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/3416)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC has three simple goals:
+
+1. Allow adding descriptions to features in `Cargo.toml`
+2. Allow specifying additional `rustdoc` configuration (favicon URL, playground
+   URL, etc) in either `Cargo.toml` or a new `rustdoc.toml`
+3. By combining the two, allow `rustdoc` will be able to document cargo
+   features, and have room to expand its configuratino options.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, <http://docs.rs> provides a simple view of available feature flags on
+a rather simple page: for example,
+<https://docs.rs/crate/tokio/latest/features>. It is helpful as a quick overview
+of available features, but means that users must manually maintain a feature
+table if they want them to be documented somehow.
+
+The second problem is that `rustdoc` has some per-crate configuration settings,
+such as relevant URLs, that are awkward to define in Rust source files using
+attributes. It is expected that there may be further configuration options in
+the future.
+
+This RFC provides a way to solve both problems: it will give `rustdoc` the
+ability to document features, and provide room to grow with more configration...
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Usage is simple: features will be able to be specified in a table (inline or
+separate) with the keys `doc`, `public`, and `requires`. Sample `Cargo.toml`:
+
+```toml
+# Cargo.toml
+
+[features]
+# current configuration
+foo = []
+# Add a description to the feature
+bar = { requires = ["foo"], doc = "simple docstring here"}
```

Comment:
It's good that you can explicitly specify it via a string, but ideally one would also be able to specify it via comments right above similar to how it works in Rust.

So:

```toml
## Does bar
bar = ["foo"]
```

Would be equivalent to:

```Rust
bar = { requires = ["foo"], doc = "Does bar"}
```

Reply:
I put this under future possibilities since I don't think it should block the main chunk of this RFC, feel free to edit what it says if you'd like